src/drakvuf.cpp

```diff
@@ -254,3 +261,38 @@ injector_status_t drakvuf_c::inject_cmd(vmi_pid_t injection_pid,
     cleanup_timer(this, timeout_thread);
     return rc;
 }
+
+struct termination_info
+{
+    std::map<vmi_pid_t, bool>& proc;
```

Comment:
This is a non-owning non-nullable pointer, which is stored. I'm **strongly** against that (it's too hard to maintain the code after and too easy to break something unintentionally by changing lifetimes). Please change it to normal pointer or make this the owner or use shared_ptr.

Reply:
Have switched to `std::shared_ptr<const...`